config/templates/metasploit-framework-wrappers/msfwrapper.erb

```diff
@@ -37,7 +37,7 @@ check_db() {
 }
 
 check_path() {
-  if [ "`which msfconsole`" = "" ]; then
+  if [ ! hash $cmd 2>/dev/null ]; then
```

Comment:
I tried it, but it had a negative effect on readability IMO. Gonna leave this as is, thanks. This function gets run exactly once ever, so it's not something we have to optimize.

```
--- a/config/templates/metasploit-framework-wrappers/msfwrapper.erb
+++ b/config/templates/metasploit-framework-wrappers/msfwrapper.erb
@@ -37,16 +37,17 @@ check_db() {
 }

 check_path() {
-  if [ ! hash $cmd 2>/dev/null ]; then
-    while true; do
-      read -p "Would you like to add $cmd and other programs to your default PATH? " yn
-        case $yn in
-          [Yy]* ) add_path; break;;
-          [Nn]* ) break;;
-              * ) echo "Please answer yes or no.";;
-        esac
-      done
+  if hash $cmd 2>/dev/null; then
+    return
   fi
+  while true; do
+    read -p "Would you like to add $cmd and other programs to your default PATH? " yn
+    case $yn in
+      [Yy]* ) add_path; break;;
+         [Nn]* ) break;;
+          * ) echo "Please answer yes or no.";;
+    esac
+  done
 }

 start_db() {
```